src/spikeinterface/sortingcomponents/waveforms/savgol_denoiser.py

```diff
@@ -16,10 +16,10 @@ def __init__(
     ):
         super().__init__(recording, return_output=return_output, parents=parents)
         self.order = order
-        if self.parents is None or not (len(self.parents) == 1 and isinstance(self.parents[0], WaveformExtractorNode)):
-            exception_string = f"SavGolDenoiser should have a single {WaveformExtractorNode.__name__} in its parents"
-            raise TypeError(exception_string)
-        waveform_extractor = self.parents[0]
+
+        waveform_extractor = find_parent_of_type(self, WaveformRetrieverNode)
```

Comment:
We need to find another name, this is ambiguous since it seems like waveforms have already been saved and only retrieved (like the new `PeakRetriever`)

@h-mayorquin what about `WaveformsNode`? I think it's general enough, it has the concept of waveforms and of being a node. Ok?

Reply:
I pushed the change, and extended the docstring for the base class ;)